### PR TITLE
Added support for Amazon

### PR DESCRIPTION
**Pull request by vicaroni**

### CODE REVIEW

File: adapters/models.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import Column, String, DateTime, Float
+
+
+Base = declarative_base()
+
+
+class EbayItem(Base):
```
Comment (Girbons) [inline adapters/models.py]: there are a lot of common fields, you could define a base model

File: babalooo.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    '''Asynchronous search for product. Saves items found in database'''
+    logging.info('ebay: searching for %s' % product)
+    response = await find_advanced(product)
+    response = response.get('findItemsAdvancedResponse')
+    if response is not None:
+        logging.info('ebay: found %s' % product)
+        items = ebay_make_items(response, count)
+        conn.process(items, 'ebay')
+    else:
+        fail = 'ebay: failed search for %s' % product
```
Comment (Girbons) [inline babalooo.py]: define a base fail message like 
```python
ERROR_MSG = '%s: failed search for %s'

# then you can do
error = ERROR_MSG % ('ebay', product)
```

File: babalooo.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    logging.info('amazon: searching for %s' % product)
+    keys = open('keys').read().split('\n')
+    amazon = bottlenose.Amazon(keys[0], keys[1], keys[2], Region=keys[3])
+    response = amazon.ItemSearch(Keywords=product['keywords'], SearchIndex='All', ResponseGroup='Medium')
+    items = amazon_make_items(response, count)
+    if items is not None:
+        logging.info('amazon: found %s' % product)
+        conn.process(items, 'amazon')
+    else:
+        fail = 'amazon: failed search for %s' % product
```
Comment (Girbons) [inline babalooo.py]: same as above

**Review by Girbons — DISMISSED** [review]

File: babalooo.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+def amazon_make_items(response, count):
+    '''
+    Constructs a list of dict objects containing information about items
+    Amazon returns search results as XML to be parsed using BeautifulSoup
+    '''
+    soup = BeautifulSoup(response, 'xml')
+    if soup.find('ItemSearchResponse') is not None:
+        items = []
+        for item in soup('Item')[:count]:
```
Comment (silvanocerza) [inline babalooo.py]: This `for` is ugly, find another solution.

File: babalooo.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        if isinstance(elem, list):
+            products_list += elem
+        elif isinstance(elem, dict):
+            products_list.append(elem)
+    return products_list
+
+
+FOUND_MSG = '%s: found %s'
+SEARCH_MSG = '%s: searching for %s'
+ERROR_MSG = '%s: failed search for %s'
```
Comment (silvanocerza) [inline babalooo.py]: It's good practice to place constants on top.

File: ebayaiohttp.py [hunk 6]
```diff
@@ -10,7 +10,7 @@
     params = {
         'OPERATION-NAME': 'findItemsAdvanced',
         'SERVICE-VERSION': '1.0.0',
-        'SECURITY-APPNAME': 'Riccardo-ebay-PRD-4c6cc25dc-e28c4d3d',
+        'SECURITY-APPNAME': open('keys').read().split('\n')[4],
```
Comment (silvanocerza) [inline ebayaiohttp.py]: Use an env var.

**Review by silvanocerza — APPROVED** [review]

**Pull request merged**